From f69ea96257d7ff6b1a90e63e6597f0c49e4b0288 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Sun, 29 Apr 2012 09:04:02 -0400
Subject: [PATCH 2/3] daemon: Add some more PAM g_debug() prints

Hopefully helpful later for whoever has to debug PAM stuff.

https://bugzilla.gnome.org/show_bug.cgi?id=675086
---
 daemon/gdm-session-worker.c |   11 ++++++++---
 1 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/daemon/gdm-session-worker.c b/daemon/gdm-session-worker.c
index 57f1071..28f0cb0 100644
--- a/daemon/gdm-session-worker.c
+++ b/daemon/gdm-session-worker.c
@@ -1084,7 +1084,10 @@ gdm_session_worker_initialize_pam (GdmSessionWorker *worker,
 
         g_assert (worker->priv->pam_handle == NULL);
 
-        g_debug ("GdmSessionWorker: initializing PAM");
+        g_debug ("GdmSessionWorker: initializing PAM; service=%s username=%s seat=%s",
+                 service ? service : "(null)",
+                 username ? username : "(null)",
+                 seat_id ? seat_id : "(null)");
 
         pam_conversation.conv = (GdmSessionWorkerPamNewMessagesFunc) gdm_session_worker_pam_new_messages_handler;
         pam_conversation.appdata_ptr = worker;
@@ -1273,7 +1276,8 @@ gdm_session_worker_authorize_user (GdmSessionWorker *worker,
         int error_code;
         int authentication_flags;
 
-        g_debug ("GdmSessionWorker: determining if authenticated user is authorized to session");
+        g_debug ("GdmSessionWorker: determining if authenticated user (password required:%d) is authorized to session",
+                 password_is_required);
 
         authentication_flags = 0;
 
@@ -1288,6 +1292,7 @@ gdm_session_worker_authorize_user (GdmSessionWorker *worker,
         /* it's possible that the user needs to change their password or pin code
          */
         if (error_code == PAM_NEW_AUTHTOK_REQD && !worker->priv->is_program_session) {
+                g_debug ("GdmSessionWorker: authenticated user requires new auth token");
                 error_code = pam_chauthtok (worker->priv->pam_handle, PAM_CHANGE_EXPIRED_AUTHTOK);
 
                 gdm_session_worker_get_username (worker, NULL);
@@ -1664,7 +1669,7 @@ session_worker_child_watch (GPid              pid,
 static void
 gdm_session_worker_watch_child (GdmSessionWorker *worker)
 {
-
+        g_debug ("GdmSession worker: watching pid %d", worker->priv->child_pid);
         worker->priv->child_watch_id = g_child_watch_add (worker->priv->child_pid,
                                                           (GChildWatchFunc)session_worker_child_watch,
                                                           worker);
-- 
1.7.7.6

